### PR TITLE
Make backstabs/market gardens/goomba stomp notifications less generic...

### PR DESCRIPTION
**Pull request by shadow93**
..and actually show the boss's name

Also, why hasn't the goomba stomp notifications text been moved to the
translations file already?

### CODE REVIEW

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -5828,6 +5828,9 @@ public Action:OnTakeDamage(client, &attacker, &inflictor, &Float:damage, &damage
 		new boss=GetBossIndex(client);
 		if(boss!=-1)
 		{
+			new String:bossName[768];
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I don't think this was intended.  Also, `decl` it.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -5828,6 +5828,9 @@ public Action:OnTakeDamage(client, &attacker, &inflictor, &Float:damage, &damage
...
+			FF2_GetBossSpecial(boss, bossName, sizeof(bossName));
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Don't call a FF2 native from within FF2.  Use the KV way instead.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 3]
```diff
@@ -6348,16 +6351,20 @@ public Action:OnStomp(attacker, victim, &Float:damageMultiplier, &Float:damageBo
...
 		JumpPower=0.0;
-		PrintHintText(victim, "Ouch!  Watch your head!");
-		PrintHintText(attacker, "You just goomba stomped somebody!");
+		PrintHintText(victim, "%t", "Player Goomba Stomped");
+		PrintHintText(attacker, "%t", "Boss Goomba Stomps", victim);
 		return Plugin_Changed;
 	}
 	else if(IsBoss(victim))
 	{
+		new String:bossName[768];
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Ditto here from above

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 4]
```diff
@@ -6348,16 +6351,20 @@ public Action:OnStomp(attacker, victim, &Float:damageMultiplier, &Float:damageBo
 		}
 		damageMultiplier=900.0;
 		JumpPower=0.0;
-		PrintHintText(victim, "Ouch!  Watch your head!");
-		PrintHintText(attacker, "You just goomba stomped somebody!");
+		PrintHintText(victim, "%t", "Player Goomba Stomped");
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Let's go a bit further and do something like `Player Goomba Stomped Boss` and `Boss Goomba Stomped Player`

**Comment by 50Wliu** [comment]
I'm not sure I'm a fan of the boss name change.  Maybe I just need to see it in action.

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 1 above)
Comment (shadow93) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: for the occasional boss with a ridiculously long boss name

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: 64 is the limit everywhere else in the code so it'd be pretty pointless to have just one spot with 768.

**Pull request closed without merging**